dotcom-rendering/src/components/ArticleMeta.apps.tsx

```diff
@@ -0,0 +1,351 @@
+import { css } from '@emotion/react';
+import type { ArticleFormat } from '@guardian/libs';
+import { ArticleDesign, ArticleSpecial } from '@guardian/libs';
+import {
+	between,
+	border,
+	from,
+	space,
+	until,
+} from '@guardian/source-foundations';
+import { StraightLines } from '@guardian/source-react-components-development-kitchen';
+import { interactiveLegacyClasses } from '../layouts/lib/interactiveLegacyStyling';
+import { getSoleContributor } from '../lib/byline';
+import { palette as themePalette } from '../palette';
+import type { Branding as BrandingType } from '../types/branding';
+import type { TagType } from '../types/tag';
+import {
+	metaContainer as metaContainerMargins,
+	shouldShowAvatar,
+	shouldShowContributor,
+} from './ArticleMeta';
+import { Avatar } from './Avatar';
+import { Branding } from './Branding.importable';
+import { CommentCount } from './CommentCount.importable';
+import { Contributor } from './Contributor';
+import { Dateline } from './Dateline';
+import { FollowWrapper } from './FollowWrapper.importable';
+import { Island } from './Island';
+import { SendAMessage } from './SendAMessage.importable';
+
+type Props = {
+	format: ArticleFormat;
+	pageId: string;
+	webTitle: string;
+	byline?: string;
+	tags: TagType[];
+	primaryDateline: string;
+	secondaryDateline: string;
+	branding?: BrandingType;
+	discussionApiUrl: string;
+	shortUrlId: string;
+	isCommentable: boolean;
+	ajaxUrl: string;
+	messageUs?: MessageUs;
+};
+
+const metaGridContainer = (showAvatar: boolean) => css`
+	display: grid;
+	grid-template-rows: auto;
+
+	${showAvatar
+		? css`
+				grid-template-columns: auto 1fr auto;
+				grid-template-areas:
+					'avatar byline comment-count'
+					'avatar byline comment-count'
+					'dateline dateline dateline';
+		  `
+		: css`
+				grid-template-columns: 1fr auto;
+				grid-template-areas:
+					'byline comment-count'
+					'byline comment-count'
+					'dateline dateline';
+		  `}
+`;
```

Comment:
Not a blocker, but ideally we’d keep this static CSS, and make the dynamic prop actually dynamic, [as per Emotion best practice](https://emotion.sh/docs/best-practices#use-the-style-prop-for-dynamic-styles).

Part of the beauty of using grid, though, is that you can keep a single `grid-template-areas` declaration, and if there are no items in the `avatar` area, it will be collapsed down to 0 pixels!